include/rtweekend.hpp

```diff
@@ -16,62 +18,92 @@
 #include <build_parameters.hpp>
 #include <xorshift.hpp>
 
+using real_t = float;
+
 // Constants
 
-constexpr float infinity = std::numeric_limits<float>::infinity();
-constexpr float pi = 3.1415926535897932385f;
+constexpr real_t infinity = std::numeric_limits<real_t>::infinity();
+constexpr real_t pi = 3.1415926535897932385f;
 
 // type aliases for float3 - vec, point and color
-using point = sycl::float3;
-using color = sycl::float3;
-using vec = sycl::float3;
+using real_vec = sycl::float3;
+
+using point = real_vec;
+using color = real_vec;
+using vec = real_vec;
 
 // Utility Functions
 
-inline float degrees_to_radians(float degrees) { return degrees * pi / 180.0f; }
+inline real_t degrees_to_radians(real_t degrees) {
+  return degrees * pi / 180.0f;
+}
+
+/**
+  @brief hash two vector using coordinates low bits
+
+  @param val1 first vector
+  @param val2 second vector
+  @return uint32_t 
+ */
+uint32_t toseed(vec const& val1, vec const& val2) {
+  uint32_t x1, y1, z1, x2, y2, z2;
+  std::memcpy(&x1, &val1.x(), sizeof(uint32_t));
+  std::memcpy(&y1, &val1.y(), sizeof(uint32_t));
```

Comment:
```suggestion
  std::memcpy(&y1, &val1.y(), sizeof y1);
```
everywhere.
This is clearer to use the type of the destination. Also it is shorter since you can avoid `()` with `sizeof` expression. :-)